flux_led/__main__.py

```diff
@@ -574,32 +576,68 @@ def _determineMode(self, ww_level, pattern_code):
                 mode = "color"
         elif pattern_code == 0x60:
             mode = "custom"
+        elif pattern_code == 0x41:
+            mode = "color"
         elif PresetPattern.valid(pattern_code):
             mode = "preset"
         elif BuiltInTimer.valid(pattern_code):
             mode = BuiltInTimer.valtostr(pattern_code)
 
         return mode
 
-    def update_state(self, retry=2):
+    def query_state(self, retry=2, led_type = None):
+
+        # default values
         msg = bytearray([0x81, 0x8a, 0x8b])
+        msg_len = 14
+
+        if self.protocol == 'LEDnet_wifi370' or led_type == 'LEDnet_wifi370':
+            msg =  bytearray([0xef, 0x01, 0x77])
+            msg_len = 11
+            led_type = 'LEDnet_wifi370'
+            self._use_csum = False
         try:
             self._send_msg(msg)
-            rx = self._read_msg(14)
+            rx = self._read_msg(msg_len)
```

Comment:
I really appreciate how you're taking the time to review my pull request. The improvements you're suggesting are great and I'm learning a lot, thank you!

I'm having trouble understanding this particular issue. As far as I can see, the situation you are describing won't really come up...
If the device uses the original protocol, it will not respond to the default message at all. The function _recieve_msg will timeout instead. So there is no way to differentiate between a broken device and a device that uses the original protocol based on the first query_status message bytearray([0x81, 0x8a, 0x8b]). 

Only when you fall back to the original protocol will the device send a response. And once you fall back, you may as well specifiy the expected length when it is different. 


This is mostly an optimization issue, isn't it? As in, we need 2 retries with the default protocol before we fall back to the original protocol. Is there a faster way to do this? cycle through the protocols before doing the 1st and 2nd retry of the other protocols perhaps?